lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -479,7 +480,20 @@ class JujuTopology:
 
     STUB = "%%juju_topology%%"
 
-    def __init__(self, model: str, model_uuid: str, application: str, charm_name: str):
+    def __new__(cls, *args, **kwargs):
```

Comment:
`new` may be used to override and customize class creation. I do not see this being a concern for beta release because we are preventing `JujuTopology` from being instantiated and thereby effectively removing it from the public interface. Hence if we so choose we could refactor the internal implementation of `JujuTopology` in a future commit and/or even make it public.